File: sort-colors/sort-colors.cs
public class Solution {
    public void SortColors(int[] nums) {
        
        int prev =0;
        int curr =0;
        int next = nums.Length-1;
        int tmp;
        
        while(curr<=next)
        {
            if(nums[curr]==0)
            {
                tmp = nums[prev];
                nums[prev++] = nums[curr];
                nums[curr++] = tmp;
            }
            else if(nums[curr]==2)
            {
                tmp = nums[curr];
                nums[curr] = nums[next];
                nums[next--] = tmp;
            }
            else{
                curr++;
            }
        }
        
    }
}

File: assignments/src/Sort_Colors.java
public class Sort_Colors {

    // Time: O(n), Space: O(1)
    public void sortColors(int[] nums) {

        int start = 0, end = nums.length-1, p = 0;

        while (p <= end) {

            if (nums[p] == 2) {
                swap(p, end, nums);
                end--;
            }

            else if (nums[p] == 0) {
                swap(p, start, nums);
                start++;
                p++;
            }

            else p++;
        }
    }
    void swap(int i, int j, int[] nums) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }
}